packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -230,13 +230,11 @@ class WebpackCLI {
 
         if (cli) {
             const processArguments = (options) => {
-                const coreFlagMap = flags
-                    .filter((flag) => flag.group === 'core')
-                    .reduce((accumulator, flag) => {
-                        accumulator[flag.name] = flag;
+                const coreFlagMap = coreFlags.reduce((accumulator, flag) => {
```

Comment:
We should avoid `reduce` too

Reply:
we can avoid it but creating a map helps so we don't loop over core flags for every arg supplied to the cli

Reply:
we should ideally do this from the start as exporting array of flags is expensive in computation to check whether one flag is supplied or not

Reply:
And we don't need this `we can avoid it but creating a map helps so we don't loop over core flags for every arg supplied to the cli`, please consider merging objects using `...`

Reply:
then we have problems in `cli.processArguments(coreFlagMap, options, coreConfig);` because then we'll be passing flags which are not supported in core which will return problems of unknown argument